src/extension.ts

```diff
@@ -132,24 +167,28 @@ export const activate = async (context: vscode.ExtensionContext) => {
   vscode.workspace.onDidSaveTextDocument(async (document) => {
     const filePath = document.uri.path
     const folder = vscode.workspace.getWorkspaceFolder(document.uri)
-    if (!folder) { return }
+    if (!folder) {
+      return
+    }
 
     const cli = new BaseCLIController(folder)
     const { repoConfigPath } = await cli.status()
 
     if (filePath === path.join(folder.uri.path, repoConfigPath)) {
       await loadRepoConfig(folder)
-    } else if (vscode.workspace.getConfiguration('devcycle-feature-flags').get('refreshUsagesOnSave')) {
+    } else if (
+      vscode.workspace
+        .getConfiguration('devcycle-feature-flags')
+        .get('refreshUsagesOnSave')
+    ) {
       await usagesDataProvider.refresh(folder, false)
     }
   })
 
   vscode.workspace.onDidChangeWorkspaceFolders(async (event) => {
     utils.checkForWorkspaceFolders()
-    const foldersToRefresh = [...event.added, ...event.removed]
-    await loginAndRefresh([...foldersToRefresh], true)
+    await loginAndRefreshAll(true)
```

Comment:
this changes from just the added and removed to all workspace folders to refresh the correct view on workspace folder changes

Reply:
if I logged out and only left the non-logged-in folder, it would be in a state where all the views are open but they are in an invalid state because we never were able to switch to the login view by setting `hasCredentialsAndProject` to false

Reply:
To avoid unnecessarily refreshing all the folders, we could update `loginAndRefresh` to use your `getLoggedInFolders` util (or create a similar util that returns true/false if it finds at least one logged in folder). Then we could go back to `loginAndRefresh` controlling the `hasCredentialsAndProject` property without the custom logic in `loginAndRefreshAll`